runtime/lua/vim/spellfile.lua

```diff
@@ -0,0 +1,281 @@
+local M = {}
+
+--- @class SpellfileConfig
+--- @field url string
+--- @field rtp string[]
+--- @field encoding string
+--- @field timeout_ms integer
+
+---@class SpellInfo
+---@field files string[]
+---@field key string
+---@field lang string
+---@field encoding string
+---@field dir string
+
+---@type SpellfileConfig
+M.config = {
+  url = 'https://ftp.nluug.nl/pub/vim/runtime/spell',
+  rtp = vim.opt.rtp:get(),
+  encoding = 'utf-8',
```

Comment:
The old code referenced `&encoding` because the 'encoding' option of Vim is configurable. In Nvim it's always utf-8. So I think we don't need this parameter at all.

If you want to keep the logic as-is, you can set `local encoding = 'utf-8'` at the top of this module. But it shouldn't be a public config setting.